qiskit/circuit/gate.py

```diff
@@ -52,7 +53,7 @@ def to_matrix(self) -> np.ndarray:
         """
         if hasattr(self, "__array__"):
             # pylint: disable=no-member
-            return self.__array__(dtype=complex)
+            return self.__array__(dtype=complex)  # type: ignore[attr-defined]
```

Comment:
Thanks for updating the PR! There are some ignores like this one which I think we should not add until we have complete typing checks in place. Otherwise new code, which is not type-checked and hence will not have type-ignore comments, will be inconsistent with this already fixed code.